src/localforage.js

```diff
@@ -164,41 +152,37 @@ class LocalForage {
     // Used to define a custom driver, shared across all instances of
     // localForage.
     defineDriver(driverObject, callback, errorCallback) {
-        var promise = new Promise(function(resolve, reject) {
+        const promise = new Promise(function(resolve, reject) {
             try {
-                var driverName = driverObject._driver;
-                var complianceError = new Error(
+                const driverName = driverObject._driver;
+                const complianceError = new Error(
                     'Custom driver not compliant; see ' +
                     'https://mozilla.github.io/localForage/#definedriver'
                 );
-                var namingError = new Error(
-                    'Custom driver name already in use: ' + driverObject._driver
-                );
 
                 // A driver name should be defined and not overlap with the
                 // library-defined, default drivers.
                 if (!driverObject._driver) {
                     reject(complianceError);
                     return;
                 }
-                if (isLibraryDriver(driverObject._driver)) {
-                    reject(namingError);
-                    return;
-                }
 
-                var customDriverMethods = LibraryMethods.concat('_initStorage');
-                for (var i = 0; i < customDriverMethods.length; i++) {
-                    var customDriverMethod = customDriverMethods[i];
+                const driverMethods = LibraryMethods.concat('_initStorage');
+                for (let i = 0, len = driverMethods.length; i < len; i++) {
+                    const customDriverMethod = driverMethods[i];
                     if (!customDriverMethod || !driverObject[customDriverMethod] ||
                         typeof driverObject[customDriverMethod] !== 'function') {
                         reject(complianceError);
                         return;
                     }
                 }
 
-                var setDriverSupport = function(support) {
-                    driverSupport[driverName] = support;
-                    CustomDrivers[driverName] = driverObject;
+                const setDriverSupport = function(support) {
+                    DriverSupport[driverName] = support;
+                    DefinedDrivers[driverName] = driverObject;
+                    // don't use a then, so that we can define
```

Comment:
Nice 👍 

Dunno if we did this before but that's clever.

Reply:
This was done in #692 by @gabrielmaldi, I just added the comment :)